internal/services/network/application_gateway_resource.go

```diff
@@ -4699,6 +5328,27 @@ func applicationGatewayCustomizeDiff(ctx context.Context, d *pluginsdk.ResourceD
 		}
 	}
 
+	probes := d.Get("probe").(*schema.Set).List()
+	for _, probe := range probes {
+		if probe == nil {
+			continue
+		}
+
+		v := probe.(map[string]interface{})
+		if value, ok := v["protocol"]; ok {
+			protocol := value.(string)
+			if protocol == string(network.ApplicationGatewayProtocolHTTP) || protocol == string(network.ApplicationGatewayProtocolHTTPS) {
+				if pathValue, exists := v["path"]; !exists || len(pathValue.(string)) == 0 {
+					return fmt.Errorf("`path` of `probe` should be specified when `protocol` of `probe` is `Http` or `Https`")
+				}
+			} else {
+				if value, exists := v["path"]; exists && len(value.(string)) > 0 {
```

Comment:
This wants to be:
```suggestion
				if v["path"] !="" && protocol != "" {
```

As otherwise, the `exists` is always true.